src/Paket.Core/PublicAPI.fs

```diff
@@ -53,27 +55,38 @@ type Dependencies(dependenciesFileName: string) =
         else
             DependenciesFile(dependenciesFileName, InstallOptions.Default, [], [], []).Save()
         Dependencies(dependenciesFileName)
-        
+    
     /// Converts the solution from NuGet to Paket.
-    static member ConvertFromNuget(force: bool,installAfter: bool,initAutoRestore: bool,credsMigrationMode: string option) : unit =        
-        let credsMigrationMode = credsMigrationMode |> Option.map NuGetConvert.CredsMigrationMode.Parse
-        let dependencies = 
-            try Some <| Dependencies.Locate()
-            with _ -> None 
-        
-        let existingDependenciesFile = 
-            if force 
-            then dependencies |> Option.map (fun d -> d.DependenciesFile)
-            else dependencies |> Option.map (fun d -> failwithf "%s already exists, use --force to overwrite" d.DependenciesFile)
-        
-        let dependenciesFileName = 
-            match existingDependenciesFile with
-            | Some file -> file
-            | None -> Path.Combine(Environment.CurrentDirectory, Constants.DependenciesFileName)
+    static member ConvertFromNuget(force: bool,installAfter: bool,initAutoRestore: bool,credsMigrationMode: string option) : unit =
+        let currentDirectory = DirectoryInfo(Environment.CurrentDirectory)
+        let rootDirectory = defaultArg (PaketEnv.locatePaketRootDirectory(currentDirectory)) currentDirectory
 
         Utils.RunInLockedAccessMode(
-            Path.GetDirectoryName(dependenciesFileName),
-            fun () ->  NuGetConvert.ConvertFromNuget(dependenciesFileName, force, installAfter, initAutoRestore, credsMigrationMode))
+            rootDirectory.FullName,
+            fun () ->
+                NuGetConvert.convertR rootDirectory force credsMigrationMode
+                |> either 
+                    (NuGetConvert.replaceNugetWithPaket initAutoRestore installAfter) 
+                    (List.iter (string >> Logging.traceError))
+        )
+
+     /// Converts the current package dependency graph to the simplest dependency graph.
+    static member Simplify(): unit = Dependencies.Simplify(false)
+
+    /// Converts the current package dependency graph to the simplest dependency graph.
+    static member Simplify(interactive : bool) =
+        match PaketEnv.locatePaketRootDirectory(DirectoryInfo(Environment.CurrentDirectory)) with
+        | Some rootDirectory ->
+            Utils.RunInLockedAccessMode(
+                rootDirectory.FullName,
+                fun () -> 
+                    PaketEnv.fromRootDirectory rootDirectory
+                    >>= Simplifier.ensureNotInStrictMode
+                    >>= Simplifier.simplify interactive
+                    |> either (Simplifier.updateEnvironment) (List.iter (string >> Logging.traceError))
```

Comment:
Cool.
Use whatever works best for you.
I think plain issue is probably enough.
On Jan 6, 2015 11:17 AM, "Tomasz Heimowski" <notifications@github.com>
wrote:

> In src/Paket.Core/PublicAPI.fs
> <https://github.com/fsprojects/Paket/pull/476#discussion-diff-22516666>:
>
> > +        )
> > +
> > +     /// Converts the current package dependency graph to the simplest dependency graph.
> > +    static member Simplify(): unit = Dependencies.Simplify(false)
> > +
> > +    /// Converts the current package dependency graph to the simplest dependency graph.
> > +    static member Simplify(interactive : bool) =
> > +        match PaketEnv.locatePaketRootDirectory(DirectoryInfo(Environment.CurrentDirectory)) with
> > +        | Some rootDirectory ->
> > +            Utils.RunInLockedAccessMode(
> > +                rootDirectory.FullName,
> > +                fun () ->
> > +                    PaketEnv.fromRootDirectory rootDirectory
> > +                    >>= Simplifier.ensureNotInStrictMode
> > +                    >>= Simplifier.simplify interactive
> > +                    |> either (Simplifier.updateEnvironment) (List.iter (string >> Logging.traceError))
>
> Will do, hopefully tonight - also will prepere a list of commands with
> status of what needs to be still converted to ROP.
> Prefer project milestone or a plain list in an issue?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fsprojects/Paket/pull/476/files#r22516666>.
>